File: ProgettoMobidev_iOS/progettomobidev/progettomobidev/FontUtility.m
//
//  FontUtility.m
//  Travel Notes
//
//  Created by gianpaolo on 04/11/16.
//  Copyright © 2016 capraraedefrancescosoft. All rights reserved.
//

#import "FontUtility.h"

@implementation FontUtility{
    NSString *titleFont;
    NSString *noteFont;
    NSString *extraFont;
}

+ (FontUtility*)getInstance
{
    static FontUtility *sharedFonts = nil;
    @synchronized(self) {
        if (sharedFonts == nil)
            sharedFonts = [[self alloc] init];
    }
    return sharedFonts;
}

- (FontUtility*)init{
    if (self = [super init]){
        titleFont = @"AccanthisADFStd-Bold";
        noteFont = @"AmmysHandwriting";
        extraFont = @"AccanthisADFStd-Regular";
    }
    return self;
}

- (UIFont*) getFont: (FontType) type andSize:(NSInteger)size{
    UIFont *font;
    switch (type) {
        case TEXT:
            font = [UIFont fontWithName:noteFont size:size];
            break;
        case TITLE:
            font = [UIFont fontWithName:titleFont size:size];
            break;
        case EXTRA:
            font = [UIFont fontWithName:extraFont size:size];
            break;
    }
    
    return font;
}

@end

File: ProgettoMobidev/app/src/main/java/com/capraraedefrancescosoft/progettomobidev/utilities/FontUtility.java
package com.capraraedefrancescosoft.progettomobidev.utilities;

import android.content.Context;
import android.graphics.Typeface;

/**
 * Created by Ianfire on 27/09/2016.
 */
public class FontUtility {

    public enum FontType{
        TITLE, EXTRA, NOTE
    }
    private Typeface title_font, extra_font, note_font;
    private static FontUtility instance;

    private FontUtility(Context context) {
        title_font = Typeface.createFromAsset(context.getAssets(), "fonts/accanthis-adf-std.bold.otf");
        extra_font = Typeface.createFromAsset(context.getAssets(), "fonts/accanthis-adf-std.regular.otf");
        note_font = Typeface.createFromAsset(context.getAssets(), "fonts/ammys_handwriting.ttf");
    }

    public synchronized static FontUtility getInstance(Context context) {
        if(instance == null)
            instance = new FontUtility(context);
        return instance;
    }

    public Typeface getFont(FontType type){
        Typeface font = null;
        switch (type){
            case TITLE:
                font = title_font;
                break;
            case EXTRA:
                font = extra_font;
                break;
            case NOTE:
                font = note_font;
                break;
        }

        return font;
    }

}
